moler/cmd/unix/sudo.py

```diff
@@ -146,10 +131,30 @@ def _validate_start(self, *args, **kwargs):
         :param kwargs: kwargs passed to super _validate_start
         :return: Nothing
         """
+        super(Sudo, self)._validate_start(*args, **kwargs)
+        if self.cmd_object and self.cmd_class_name:
+            self.set_exception(CommandFailure(self,
```

Comment:
inside _validate_start() we should `raise` instead of `self.set_exception()` - there is context manageget that will `.set_exception()` for non-main-thread